Revert "[new config options][the eternal playtest] Mulligans/Continuous ...

MrsTonedOne opened the pull request:
Reverts tgstation/-tg-station#9149

Changes that require people copy over config options to their server's custom config to restore normal functionality are getting really old, really fucking quickly.

Right now, if you have a test server set to track all folders but config from git, and you run, the round will automatically end because it defaults to extended ending if all antags are dead without config modifcations.

MrsTonedOne committed:
Revert "[new config options][the eternal playtest] Mulligans/Continuous for any roundtype"

CorruptComputer commented:
So because you are lazy you're going to revert a good feature?

MrStonedOne commented:
Explain to me how extended automatically ending in any config scenario is a good idea?

CorruptComputer commented:
Change the default, don't remove the entire feature. And we should look at changing the config to be more modular,  I agree that it is annoying to redo everything, but that's just something you have to do when someone touches that file.

MrStonedOne commented:
This is the third time incoming has changed the config in a way that broke servers that didn't update their config.

This is also more of a punishment to make them redo the change sanely. Since they just don't learn.

MrStonedOne commented:
>"but that's just something you have to do when someone touches that file."

See, its easy as fuck to factor any change to the config file in a way to allow for old behavior if the changed config option is not in the file. I've added and changed things in the config file mutiple times. all of them having NO CHANGE if the new config option wasn't present.

Dorsisdwarf commented:
@CorruptComputer , it's not "because he's lazy", it's "Because this screws with the default settings for any downstreams too"

CorruptComputer commented:
If something isn't found in the config there should be a way to automatically write the option in and notify the host that the config has been updated.

MrStonedOne commented:
Can't be done, too many things are disabled by removing them from the config.

So how do you disable them would getting spammed with notifcations?

CorruptComputer commented:
Change the way we disable them, instead of commenting them out have it as something like:

!continuous_rounds

Iamgoofball commented:
For fuck's sake why don't we just update the repo config and put a notice up to update your Config files

CorruptComputer commented:
@Iamgoofball :+1:

Incoming5643 commented:
The config was updated within hours of this issue becoming apparent because it was so obvious. I moved to have Ikkarus and you fix this immediately upon me hearing about the config file not being up to date. The default config has these options included, so anyone downloading a copy does not run into this issue. Anyone who actually keeps the config file up to date would not have noticed any problems. I just didn't know that you didn't update your game_options.txt unless prompted.

It was my bad for not giving you more explicit warning that new config options were available, I thought you'd see the [new config options] on github. In the future I'll be sure to give you more explicity warning.

That said there shouldn't be any more need to touch the configs for mulligans after this.

The reason there are config setting for seemingly insane settings like "non-continuous traitor" is to assure that you have absolute control over the system. It covers all potentialities, even the ones most people probably wouldn't care about. Server owners can tweak it any which way they want to control its effects. I'm adding all these config options for your benefit, and it was a big design goal for the project.

It's difficult to set "expected behavior" for these things for when configs are missing because what people consider "normal" varies depending on server culture. Some places end with wizard death all the time, some never do. Which is "the normal way" there?

Like goof said a more proactive in code way to alert owners to update their config files would be a good thing.

In the future please just send me a PM or ask me to get on IRC or make an issue report first. At the very least @incoming5643 me here.

Ikarrus commented:
inb4 headmins disable mulligans again

Incoming5643 commented:
Please don't. We're so fucking close to full functionality, and the new pull is working great. Mulligan has never been LESS fucked.

Ikarrus commented:
I'm just saying they already did before I overrid the config with the last update. They seem pretty fed up with it and they might end up asking to disable them again.

Incoming5643 commented:
The era of big intrusive bugs with this is over once I'm sure that antag death detection works right.

This is literally the last thing that can piss people off overtly, any remaining bugs going forward would just fail to create antags, which, while a bug, isn't a crisis.

Incoming5643 commented:
Also there were bug fixes in this commit, so outright reversion will never be a good idea regardless of any underlying feelings about the feature.

If you would prefer that I outsource these options to their own txt file that you could have set to automatically update, this is something that can be done.

Incoming5643 commented:
I could also swap "CONTINUOUS" for NONCONTINUOUS so missing config options would default to round continuous instead. In retrospect this makes more sense but I wasn't considering that at the time because it had not occured to me that server owners weren't keeping the config file up to date.

Just know that if I do THAT it means you have to touch game_options one more time (unless I also outsource the options to their own .txt

optimumtact commented:
:-1:
go away server losers stop bullying us

MrStonedOne commented:
Let me back track and explain my issues with this in a bit more detail now that I've had some sleep and I'm less cranky.

1: My major source of annoyment with this, wasn't from the game servers, but from my test/dev server.

2: All config changes should take this habit into account. It really isn't hard to make all config changes have sane defaults if they aren't included. in this case. you make a new config option, called EXPLICIT_CONTINUOUS_ROUNDS that if not present, restores old behavior for rounds that don't end when antags die.

Like, if it's not, this should be a rule.

3: At no point is it acceptable that certain config options mean certain rounds will automatically end right after starting in certain conditions.

If a tator round starts, but it can't find any tators because everybody is security/captain. It won't spawn any. Your new config shit breaks those rounds. It is just bad design to have it be possible that a round can end right after starting. be it tator, changeling, extended, etc.

So there at least needs a sanity check to ensure that antags were actually spawned. Or at the worst, make it check that config when starting the game mode, and if it exists, revert back to lobby if it can't spawn antags or the mode doesn't have any.

Incoming5643 commented:
There's a sanity for that now in #9235, I'll reword the message to make it clear what could cause it.

I honestly didn't know people didn't keep their game_options up to date and it wasn't something I considered when programming this. It's not a very hard fix so I'll push it in with #9235.

Just, yeah, that's a bug of a flavor I don't usually see. If ya just told me that I'd have jumped on fixing it.

Incoming5643 commented:
#9235 should now take care of the concerns you have.
